python/cudf/cudf/_libxx/parquet.pyx

```diff
@@ -43,7 +44,7 @@ cpdef generate_pandas_metadata(Table table, index):
     # Columns
     for name, col in table._data.items():
         col_names.append(name)
-        types.append(col.to_arrow().type)
+        types.append(np_to_pa_dtype(col.dtype))
```

Comment:
Does this handle categorical typed columns as well? If not we'd need it to.

Reply:
Seems like `categorical` types aren't supported yet: https://github.com/rapidsai/cudf/blob/branch-0.13/python/cudf/cudf/io/parquet.py#L191
Added similar validation in meta generation aswell.